chainio/clients/elcontracts/writer.go

```diff
@@ -442,3 +451,325 @@ func (w *ChainWriter) ProcessClaims(
 
 	return receipt, nil
 }
+
+func (w *ChainWriter) ForceDeregisterFromOperatorSets(
+	ctx context.Context,
+	operator gethcommon.Address,
+	avs gethcommon.Address,
+	operatorSetIds []uint32,
+	waitForReceipt bool,
+) (*gethtypes.Receipt, error) {
+	if w.avsDirectory == nil {
```

Comment:
I think we should check for the `allocationManager` here rather than the `avsDirectory`

Reply:
Can't we keep the check for any of them? Like

```go
if w.avsDirectory == nil || w.allocationManager == nil {
```

That way users have the option to keep using the old workflow with the new version of the SDK